ocaml/xapi/repository.ml

```diff
@@ -28,13 +28,24 @@ let updates_in_cache : (API.ref_host, Yojson.Basic.t) Hashtbl.t =
   Hashtbl.create 64
 
 let introduce ~__context ~name_label ~name_description ~binary_url ~source_url
-    ~update ~gpgkey_path =
-  assert_url_is_valid ~url:binary_url ;
-  assert_url_is_valid ~url:source_url ;
+    ~update ~gpgkey_path ~origin =
+  let _ =
+    match origin with
+    | `remote ->
+        assert_url_is_valid ~url:binary_url ;
+        assert_url_is_valid ~url:source_url
+    | `bundle ->
+        ()
```

Comment:
For the `bundle`, the binary_url and source_url should be ensured as empty strings.